snapcraft_legacy/plugins/v2/rust.py

```diff
@@ -71,39 +105,99 @@ def get_build_snaps(self) -> Set[str]:
+        options = self.options
+        if not options.rust_channel and self._check_system_rust():
+            logger.info("Rust is installed on the system, skipping rustup")
+            return []
+
+        rust_channel = options.rust_channel or "stable"
+        if rust_channel == "none":
+            return []
+        if not self._check_rustup():
+            logger.info("Rustup not found, installing it")
+            return self._get_setup_rustup(rust_channel)
+        logger.info("Switch rustup channel to %s", rust_channel)
+        return [
+            f"rustup update {rust_channel}",
+            f"rustup default {rust_channel}",
         ]
 
-        if self.options.rust_features:
-            cmd.extend(
-                ["--features", "'{}'".format(" ".join(self.options.rust_features))]
+    def get_build_commands(self) -> List[str]:
+        options = self.options
+
+        rust_build_cmd: List[str] = []
+        config_cmd: List[str] = []
+
+        if options.rust_features:
+            features_string = " ".join(options.rust_features)
+            config_cmd.extend(["--features", f"'{features_string}'"])
+
+        if options.rust_use_global_lto:
+            logger.info("Adding overrides for LTO support")
+            config_cmd.extend(
+                [
+                    "--config 'profile.release.lto = true'",
+                    "--config 'profile.release.codegen-units = 1'",
+                ]
             )
 
-        return " ".join(cmd)
-
-    def get_build_commands(self) -> List[str]:
-        return [self._get_rustup_command(), self._get_install_command()]
+        if options.rust_no_default_features:
+            config_cmd.append("--no-default-features")
+
+        for crate in options.rust_path:
+            logger.info("Generating build commands for %s", crate)
+            config_cmd_string = " ".join(config_cmd)
+            # TODO(liushuyu): the current fallback installation method will also install
+            # compiler plugins into the final Snap, which is likely not what we want.
+            # pylint: disable=line-too-long
+            rust_build_cmd_single = dedent(
+                f"""\
+                if cargo read-manifest --manifest-path "{crate}"/Cargo.toml > /dev/null; then
+                    cargo install -f --locked --path "{crate}" --root "${{SNAPCRAFT_PART_INSTALL}}" {config_cmd_string}
+                    # remove the installation metadata
+                    rm -f "${{SNAPCRAFT_PART_INSTALL}}"/.crates{{.toml,2.json}}
+                else
+                    # virtual workspace is a bit tricky,
+                    # we need to build the whole workspace and then copy the binaries ourselves
+                    pushd "{crate}"
+                    cargo build --workspace --release {config_cmd_string}
+                    # install the final binaries
+                    # TODO(liushuyu): this will also install proc macros in the workspace,
```

Comment:
Can you remove `TODO(liushuyu):` from the code and create a [task](https://github.com/snapcore/snapcraft/issues/new/choose) in github?  Your comment [here](https://github.com/snapcore/snapcraft/pull/4297/files#r1321710625) is useful context to add to the task.

We are slowly removing all `todo` and `fixme` comments from our codebase and using github issues instead.

Reply:
Done in #4364 4364